### PR TITLE
feat: add forge prime and bounded orientation

### PR DESCRIPTION
**Pull request by harshanandak**
## Problem
D21/D22 need a JSON-first agent entry surface for bounded project orientation, but Forge did not yet expose `forge orient`, `forge prime`, or an issue-scoped `forge recap <issue>` contract.

## Root Cause
The existing `forge recap` command only implements the project-wide activity recap over Beads JSONL. The D21 bounded context contract was accepted in docs but had no deterministic file-assembly implementation or token-budgeted JSON output.

## Fix
Adds deterministic file assembly V1 for orientation/recap without implementing full FTS5 KnowledgeStore search. The new assembler reads project identity, `docs/PROJECT_DESIGN.md` headline decisions, active work-folder artifacts, Kernel-replaceable ready/claim placeholders, and Beads issue JSONL only as compatibility for issue lookup. Adds `forge orient --json`, `forge prime --json`, and additive `forge recap <issue> --json` while preserving no-arg recap behavior.

## Value
Agents get bounded, source-backed session context with `next_commands[]`, explicit token budgets, deterministic truncation, and source classifications that can be swapped to Kernel-backed reads later.

## Beads
Closes forge-2agy.9.4

<details>
<summary>Implementation Details</summary>

### Test Coverage
- Tests: 32 focused tests passing locally; pre-push ran 18 targeted tests and 266 affected edge-case tests passing.
- Scenarios covered: token budget/truncation, `next_commands[]`, `forge orient --json`, `forge prime --json`, issue-scoped `forge recap <issue> --json`, no-arg recap compatibility, command registry/exports.

### Security Review
- OWASP Top 10: new code is read-only local file assembly; it does not add auth, network calls, shell execution, or write paths. Beads is read only through `.beads/issues.jsonl` compatibility projection for issue lookup.
- Automated scan: pre-push lint passed; affected edge-case security tests passed in the pre-push hook.

### Design Doc
See:
- `docs/work/2026-06-06-kernel-backlog-memory-roadmap/decisions.md` D21-D22
- `docs/work/2026-06-06-kernel-backlog-memory-roadmap/agent-interface-layer.md`
- `docs/PROJECT_DESIGN.md` PD-20260611-orient-recap-file-assembly and PD-20260611-agent-interface-parity

### Key Decisions
- D21 V1 is deterministic file assembly, not full FTS5/KnowledgeStore search.
- No-arg `forge recap` keeps the existing project-wide activity recap contract.
- `forge recap <issue>` is additive and source-backed with Beads compatibility reads only.
- Ready queue and active claims are explicit placeholders so Kernel can replace them without changing the JSON shape.
- PR is draft because the Prompt 1 dependency was not verified as merged in this checkout/GitHub search.

### Documentation Updated
None - implementation follows the already accepted D21/D22 docs.

### Validation
- [x] Type check passing
- [x] Lint passing (0 errors, 0 warnings)
- [x] All tests passing
- [x] Security review completed

</details>

---

### Self-Review Checklist

- [ ] I reviewed the full diff on GitHub
- [x] No debug code (console.log, commented code, temporary changes)
- [x] ESLint passes with zero warnings (`bunx eslint .` / pre-push lint)
- [x] All tests pass locally (`bun test` focused + pre-push targeted/affected tests)
- [x] No hardcoded secrets or API keys
- [x] No breaking changes (or documented in migration guide)
- [x] TDD compliance: All source files have corresponding tests

## Summary by CodeRabbit

* **New Features**
  * Added `forge orient` to generate bounded project orientation output with token-budget limiting and optional `--json`.
  * Added `forge prime` to provide session-oriented output derived from the bounded orientation, with `--json`.
  * Enhanced `forge recap` with an optional positional `[issue]` to produce issue-scoped recaps (supports `--json` and `--budget`).
* **Tests**
  * Added Bun test coverage for `orient`, `prime`, and `recap`, including budget/truncation and next-command output contracts.
  * Updated a release readiness expectation.
* **Documentation**
  * Updated a runtime call-site checklist doc.

### CODE REVIEW

File: lib/orientation.js [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+  if (targetChars <= 32) {
+    return { content: '', truncated: true };
+  }
+  const suffix = '\n[truncated deterministically by token budget]';
+  const nextContent = String(content).slice(0, Math.max(0, targetChars - suffix.length)).trimEnd();
+  return {
+    content: `${nextContent}${suffix}`,
+    truncated: true,
+  };
+}
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/orientation.js:352]: ✅ Fixed in f8f5ed4: `truncateText` now fits the truncation marker into the allowance. If even the marker exceeds `targetTokens` the section is dropped (returns empty), and otherwise content is shrunk until the marker-inclusive result is within the cap — so a small allowance can no longer emit a larger-than-allowed section, and `token_budget.used` no longer hides an overage. Existing budget tests (used ≤ requested) still pass.
